stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -467,25 +466,21 @@ public class ChatClient internal constructor(
         offset: Int,
         limit: Int,
         type: String
-    ): Call<List<AttachmentWithDate>> {
+    ): Call<List<Attachment>> =
+        getMessagesWithAttachments(channelType, channelId, offset, limit, type).map { messages ->
```

Comment:
As far as I remember this returns messages with all attachments when at least one of them has the type that we wanted, so if in some channel, we have a message with image and two files, and we want to get only `imageAttachments` this won't work correctly. We will get this message with all attachments and just map them, so our result we will get one image and two files. Shouldn't we filter only the desired attachments type?